diff --git a/include/wasmtime.hh b/include/wasmtime.hh
index 57e2a95..e0d6b9b 100644
--- a/include/wasmtime.hh
+++ b/include/wasmtime.hh
@@ -1746,6 +1746,8 @@ class Store {
   Context context() { return this; }
 };
 
+class Val;
+
 /**
  * \brief Representation of a WebAssembly `externref` value.
  *
@@ -1779,7 +1781,8 @@ public:
   /// Note that `val` should be safe to send across threads and should own any
   /// memory that it points to. Also note that `ExternRef` is similar to a
   /// `std::shared_ptr` in that there can be many references to the same value.
-  template <typename T>
+  template <typename T,
+            typename = std::enable_if_t<!std::is_same_v<std::decay_t<T>, Val>>>
   explicit ExternRef(T val)
       : ExternRef(wasmtime_externref_new(
             std::make_unique<std::any>(std::move(val)).release(), finalizer)) {}
